File: tl/src/main/java/com/github/badoualy/telegram/tl/api/TLInputMessagesFilterDocument.java
package com.github.badoualy.telegram.tl.api;

/**
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
public class TLInputMessagesFilterDocument extends TLAbsMessagesFilter {
    public static final int CONSTRUCTOR_ID = 0x9eddf188;

    public TLInputMessagesFilterDocument() {
    }

    @Override
    public String toString() {
        return "inputMessagesFilterDocument#9eddf188";
    }

    @Override
    public int getConstructorId() {
        return CONSTRUCTOR_ID;
    }

    @Override
    @SuppressWarnings("PointlessBooleanExpression")
    public boolean equals(Object object) {
        if (!(object instanceof TLInputMessagesFilterDocument)) return false;
        if (object == this) return true;

        TLInputMessagesFilterDocument o = (TLInputMessagesFilterDocument) object;

        return true;
    }
}

File: tl/src/main/kotlin/com/github/badoualy/telegram/tl/api/TLInputMessagesFilterDocument.kt
package com.github.badoualy.telegram.tl.api

/**
 * inputMessagesFilterDocument#9eddf188
 *
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
class TLInputMessagesFilterDocument : TLAbsMessagesFilter() {
    private val _constructor: String = "inputMessagesFilterDocument#9eddf188"

    override val constructorId: Int = CONSTRUCTOR_ID

    override fun toString() = _constructor

    override fun equals(other: Any?): Boolean {
        if (other !is TLInputMessagesFilterDocument) return false
        if (other === this) return true

        return true
    }
    companion object  {
        const val CONSTRUCTOR_ID: Int = 0x9eddf188.toInt()
    }
}
